storm-core/src/jvm/org/apache/storm/scheduler/blacklist/BlacklistScheduler.java

```diff
@@ -0,0 +1,212 @@
+package org.apache.storm.scheduler.blacklist;
+
+import org.apache.storm.Config;
+import org.apache.storm.scheduler.*;
+import org.apache.storm.scheduler.blacklist.reporters.IReporter;
+import org.apache.storm.scheduler.blacklist.strategies.IBlacklistStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Created by howard.li on 2016/6/29.
+ */
+public class BlacklistScheduler implements IScheduler {
+    private static final Logger LOG = LoggerFactory.getLogger(BlacklistScheduler.class);
+    DefaultScheduler defaultScheduler;
+    @SuppressWarnings("rawtypes")
+    private Map _conf;
+
+    private int toleranceTime;
+    private int toleranceCount;
+    private int resumeTime;
+    private IReporter reporter;
+    private IBlacklistStrategy blacklistStrategy;
+
+    private int nimbusMonitorFreqSecs;
+
+
+    private Map<String,Set<Integer>> cachedSupervisors;
+
+    //key is supervisor key ,value is supervisor ports
+    private CircularBuffer<HashMap<String,Set<Integer>>> toleranceBuffer;
+
+    @Override
+    public void prepare(Map conf) {
+        LOG.info("prepare black list scheduler");
+        LOG.info(conf.toString());
+        defaultScheduler=new DefaultScheduler();
+        defaultScheduler.prepare(conf);
+        _conf=conf;
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME)){
+            toleranceTime=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME);
+        }
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT)){
+            toleranceCount=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT);
+        }
+        if(_conf.containsKey(Config.BLACKLIST_SCHEDULER_RESUME_TIME)){
+            resumeTime=(Integer)_conf.get(Config.BLACKLIST_SCHEDULER_RESUME_TIME);
+        }
```

Comment:
Should we use [Utils.getInt](https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/utils/Utils.java#L753) or [Utils.getInt](https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/utils/Utils.java#L792) to avoid `Null` being assigned.